src/docfx/build/Build.cs

```diff
@@ -27,49 +27,56 @@ public static async Task Run(string docsetPath, CommandLineOptions options, Repo
             var docset = new Docset(context, docsetPath, config, options).GetBuildDocset();
             var monikersProvider = new MonikersProvider(docset);
 
-            // TODO: toc map and xref map should always use source docset?
-            var tocMap = BuildTableOfContents.BuildTocMap(context, docset, monikersProvider);
-            var xrefMap = XrefMap.Create(context, docset, metadataProvider, monikersProvider);
+            using (var gitCommitProvider = new GitCommitProvider())
```

Comment:
Will this cause more memory consumption since the scope of `gitCommitProvider` is bigger? @OsmondJiang had this issue before

Reply:
That one is tracked by #3720

Reply:
no, actually original code, convert to legacy will not have any memory presses